consensus/types/src/beacon_state.rs

```diff
@@ -2478,9 +2478,12 @@ impl<E: EthSpec> BeaconState<E> {
     }
 
     pub fn compute_merkle_proof(&self, generalized_index: usize) -> Result<Vec<Hash256>, Error> {
```

Comment:
The API of this function is quite strange, as it appears like a generic function but only allows to generate proofs of a very specific and small set of `generalized_index`. Maybe instead we should make this one private and expose
```
compute_current_sync_committee_proof()
compute_next_sync_committee_proof()
compute_finalized_root_proof()
```

Reply:
updated api [9341d87](https://github.com/sigp/lighthouse/pull/6361/commits/9341d87127e1d99ea841e1321a0ed1e5faa5e76a)

I've introduced two public methods

```
compute_sync_committee_proof
compute_finalized_root_proof
```

since current and next sync committee proof logic is nearly identical (minus the generalized indices) I combined them into one method.